watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -248,6 +361,28 @@ def display_results(m):
     for u in unit_list:
         m.fs.dye_separation.component(u).report()
 
+    if hasattr(m.fs, "gac"):
+        m.fs.costing = WaterTAPCosting()
+        m.fs.costing.base_currency = pyunits.USD_2021
```

Comment:
Only skimming changed code and not the whole body of code here, but I would imagine that creating a costing block and setting a base currency would be something you'd do with or without GAC. I suppose this may be due to the rest of the flowsheet only having ZO models and costing and this is the first instance of using a WaterTAPCosting block for a "detailed" model.